react-form-controls/src/NewsletterForm.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label htmlFor="email">Email</label>
+        <input id="email" type="email" onChange={this.handleChange} value={this.state.value}></input>
+        <button>Sign Up</button>
+      </form>
+    );
+  }
+
+  handleChange() {
+    this.setState(prevState => (
+      { value: event.target.value }
+    ));
+  }
+
+  handleSubmit() {
+    console.log(this.state);
+    event.preventDefault();
```

Comment:
Although the `event` object is briefly global, it's best to always include an `event` parameter for event handler functions/methods